internal/workflow.go

```diff
@@ -1862,6 +1875,15 @@ func WithDataConverter(ctx Context, dc converter.DataConverter) Context {
 	return ctx1
 }
 
+// WithPriority adds a priority to the context.
+//
+// Exposed as: [go.temporal.io/sdk/workflow.WithWorkflowPriority]
+func WithWorkflowPriority(ctx Context, priority Priority) Context {
+	ctx1 := setWorkflowEnvOptionsIfNotExist(ctx)
+	getWorkflowEnvOptions(ctx1).Priority = convertToPBPriority(priority)
+	return ctx1
+}
```

Comment:
IMO this is advanced enough that users can just set it via `WithChildOptions` (I kinda wish we did that with versioning intent too). Not sure what the bar is for each child option getting its own `With` call, but I figure these settings are surely not more popular than a `ParentClosePolicy`.